src/main/java/com/rtb/projectmanagementtool/auth/AuthOps.java

```diff
@@ -27,18 +27,25 @@ public AuthOps(DatastoreService ds) {
     this.controller = new UserController(ds);
   }
 
-  public String getAuthID() {
-    // call auth service to generate AuthID
-    UserService userService = UserServiceFactory.getUserService();
-    String AuthID = userService.getCurrentUser().getUserId();
-    return AuthID;
+  public String getAuthID(HttpServletRequest request, HttpServletResponse response) {
+    if (whichUserIsLoggedIn(request, response) != -1) {
+      // call auth service to generate AuthID
+      UserService userService = UserServiceFactory.getUserService();
+      String AuthID = userService.getCurrentUser().getUserId();
+      return AuthID;
+    }
+    // if no one logged in, return empty string
+    return "";
   }
 
   // get/create cookie and set value to userID
-  public void createAndSetCookieNewUser(HttpServletRequest request, long userID) {
+  public void createAndSetCookieNewUser(
+      HttpServletRequest request, HttpServletResponse response, long userID) {
     Cookie cookie = getCurrCookie(request);
```

Comment:
If you're overwriting the cookie anyways, no need to search for the old one and replace.  You can just build a new cookie (by filling in name and value) then dump it into the response right away.